app/jobs/people/destroy_roles_job.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2023, Schweizer Alpen-Club. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+#
+class People::DestroyRolesJob < RecurringJob
+  run_every 1.day
+
+  def perform
+    roles = Role.where('delete_on <= ?', Time.zone.today)
+    roles.find_each(&:destroy!)
```

Comment:
Wenn die Rolle archiviert ist, wird das [fehlschlagen](https://github.com/hitobito/hitobito/blob/master/app/models/role.rb#L243) mit einer Exception ActiveRecord::ReadOnlyRecord

Reply:
Wir können das verhindern, wenn wir im [beim Archivieren](https://github.com/hitobito/hitobito/blob/master/app/models/group.rb#L274) alle in der Zukunft liegenden `delete_on`s auf `nil` zurücksetzen.